docs/development/testing.rst

```diff
@@ -0,0 +1,290 @@
+        "tests.fixtures.parachutes.parachute_fixtures",
+        "tests.fixtures.rockets.rocket_fixtures",
+        "tests.fixtures.surfaces.surface_fixtures",
+        "tests.fixtures.units.numerical_fixtures",
+        "tests.fixtures.monte_carlo.monte_carlo_fixtures",
+        "tests.fixtures.monte_carlo.stochastic_fixtures",
+        "tests.fixtures.monte_carlo.stochastic_motors_fixtures",
+    ]
+
+Pay close attention to the organization of the plugins, they are also organized in a way
+to help the developer position the fixtures in a certain pattern. 
+
+To finish, let's take a quick look inside the tests directory structure. Consider the **motor**
+folder containing its fixtures:
+
+.. code-block:: rst
+
+    motor/
+    ├── __init__.py
+    ├── Cesaroni_M1670_shifted.eng
+    ├── Cesaroni_M1670.eng
+    ├── generic_motor_fixtures.py
+    ├── hybrid_fixtures.py
+    ├── liquid_fixtures.py
+    ├── solid_motor_fixtures.py
+    └── tanks_fixtures.py
+
+Observe the naming convention for the fixtures within the modules and also how the fixtures were
+structured, such that each kind of motor contains a module loaded with its needed fixtures:
+
+* kindofmotor_fixtures.py
+
+One is expected to follow this convention.
+
+Unit tests definition
+---------------------
+
+Within a complex code such as RocketPy, some definitions or agreements need to be reviewed or sophisticated
+to make sense within a projec. In RocketPy, unit tests are/can be **sociable**, which **still** means that:
+
+* (Speed) They have to be **fast**.
+* (Isolated behavior) They focus on a **small part** of the system. Here, we focus on methods.
+
+*However*, as already said, they are/can be sociable:
+
+* (Sociable) The tested unit relies on other units to fulfill its behavior.
+ 
+The classification depends on whether the test isolates the unit under test from its dependencies or allows them 
+to interact naturally. In practical terms, consider the test:
+
+.. code-block:: python
+
+    def test_evaluate_total_mass(calisto_motorless):
+        """Tests the evaluate_total_mass method of the Rocket class.
+        Both with respect to return instances and expected behaviour.
+
+        Parameters
+        ----------
+        calisto_motorless : Rocket instance
+            A predefined instance of a Rocket without a motor, used as a base for testing.
+        """
+        assert isinstance(calisto_motorless.evaluate_total_mass(), Function)
+
+This test is **sociable** because it relies on the actual Rocket instance and tests its real behavior without 
+isolating the Rocket class from its potential interactions with other classes or methods within its implementation. 
+It checks the real implementation of ``evaluate_total_mass`` rather than a mocked or stubbed version, ensuring that the 
+functionality being tested is part of the integrated system.
+
+Please note that writing an unit test which is solitary is **allowed**. The classification regarding solitary and
```

Comment:
Do you think this could confuse people writing units tests? Would say we should explain that in the docs?

Reply:
I think we can say something like "Please note that writing an unit test which is solitary is **allowed**, just make sure to back it up with proper contract tests when applicable (...) ".

Personally I'm not worries about confusion here because we took sides and shared that our preference is for sociable unit tests and went forward to explain with examples, so if one decides to go with solitary its a deliberative choice probably due to style preferences, and in this case one should know its drawbacks and make it up for it.

Reply:
Great explanation, agree